src/borg/helpers/manifest.py

```diff
@@ -86,7 +86,9 @@ def list(self, *, glob=None, match_end=r'\Z', sort_by=(), first=None, last=None,
         if isinstance(sort_by, (str, bytes)):
             raise TypeError('sort_by must be a sequence of str')
         regex = re.compile(shellpattern.translate(glob or '*', match_end=match_end))
-        archives = [x for x in self.values() if regex.match(x.name) is not None]
+        archives = [x for x in self.values() if ((regex.match(x.name) is not None) and ('checkpoint' not in x.name))]
+        if not consider_checkpoints:
+            archives = [x for x in self.values() if regex.match(x.name) is not None]
```

Comment:
1st line should build the full list into `archives` (and only do the regex match).

3rd line should only remove the checkpoint archives from the already computed `archives` list (not do the regex stuff again).